src/main/java/com/cloudbees/jenkins/support/impl/AdvancedProcFilesRetriever.java

```diff
@@ -0,0 +1,122 @@
+package com.cloudbees.jenkins.support.impl;
+
+import com.cloudbees.jenkins.support.api.Container;
+import com.cloudbees.jenkins.support.api.FilePathContent;
+import com.cloudbees.jenkins.support.util.SystemPlatform;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.FilePath;
+import hudson.model.Computer;
+import hudson.model.Node;
+import jenkins.model.Jenkins;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.DoNotUse;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+
+/**
+ * Base class for gathering specified /proc files with the capacity of selecting whether filter or not each file
+ * content. You may want to filter some files and not doing so to other files.
+ */
+public abstract class AdvancedProcFilesRetriever extends ProcFilesRetriever {
+    private static final Logger LOGGER = Logger.getLogger(AdvancedProcFilesRetriever.class.getName());
+
+    /**
+     * If you want to use this method, it's best to use {@link ProcFilesRetriever}.
+     */
+    @Override
+    @Restricted(DoNotUse.class)
+    public Map<String, String> getFilesToRetrieve() {
+        LOGGER.warning("No proc files will be included to the bundle. You should use the class ProcFilesRetriever or the method AdvancedProdFilesRetriever#getProcFilesToRetrieve");
```

Comment:
You shouldn't use this method with this class, this class is intended to be used with the new ```getProcFilesToRetrieve``` where you can define which files are filtered or not filtered.